android/app/src/main/java/com/zulipmobile/MainActivity.kt

```diff
@@ -11,6 +11,25 @@ import com.swmansion.gesturehandler.react.RNGestureHandlerEnabledRootView;
 import com.zulipmobile.notifications.*
 import com.zulipmobile.sharing.maybeHandleShareIntent
 
+/* Returns true just if we did handle the intent. */
+fun maybeHandleOpenNotificationIntent(intent: Intent): Boolean {
```

Comment:
I think we probably want this to live somewhere else, and probably have a better name? I haven't settled on anything definite for either yet though.

Reply:
Yeah. Looking at this function and now looking at the share-to version beside it, I think what I really want is to have those two bits of code next to each other -- and probably cleanest as a single `when (intent.action)`, with sub-conditions as needed (like looking for `EXTRA_NOTIFICATION_DATA`, though actually the cleanest thing there -- which can be a followup -- is probably to parse the outermost parts of the URL.)

IOW rather than align the name of `maybeHandleIntent` to reflect where it lives, align where it lives to reflect its name: move it to be a private method on `MainActivity`, and then later in the branch add the open-notification case to it.

Part of why I'd like them to be next to each other is that they're carving out two different swaths of the same space of possibilities -- if we were to have any two things like this that could overlap in what they covered, then the order we called them in would start mattering, and it'd probably be a bug. So it's helpful to have them next to each other and ideally in the same `when` statement, so that it's easy to see they're mutually exclusive and keep them that way.